### ISSUE TITLE
Module 'rook' has failed: HTTPSConnectionPool(host='172.16.0.1', port=443): Max retries exceeded with url: /api/v1/namespaces/rook-ceph/pods (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:897)'),))

### ISSUE THREAD

**Issue by daodaoyuXX** [rook/rook#14386]
bash-4.4$ ceph -v
ceph version 18.2.2 (531c0d11a1c5d39fbfe6aa8a521f023abf3bf3e2) reef (stable)
bash-4.4$ ceph status
  cluster:
    id:     c3f35661-a1fb-4858-8afb-1ddb2c462fb1
    health: HEALTH_ERR
            
[cluster.zip](https://github.com/user-attachments/files/15986240/cluster.zip)

            8 mgr modules have recently crashed

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary

**Logs to submit**:

* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

  To get logs, use `kubectl -n <namespace> logs <pod name>`
  When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
  Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Cluster Status to submit**:

* Output of kubectl commands, if necessary

  To get the health of the cluster, use `kubectl rook-ceph health`
  To get the status of the cluster, use `kubectl rook-ceph ceph status`
  For more details, see the [Rook kubectl Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/kubectl-plugin)
NAME        DATADIRHOSTPATH   MONCOUNT   AGE     PHASE   MESSAGE                        HEALTH       EXTERNAL   FSID
rook-ceph   /var/lib/rook     3          7h13m   Ready   Cluster created successfully   HEALTH_ERR              c3f35661-a1fb-4858-8afb-1ddb2c462fb1

**Environment**:
* OS (e.g. from /etc/os-release):Ubuntu 24.04 LTS
* Kernel (e.g. `uname -a`):GNU/Linux 6.8.0-35-generic x86_64
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):1.14.7
* Storage backend version (e.g. for ceph do `ceph -v`):ceph version 18.2.2
* Kubernetes version (use `kubectl version`):v1.30.1
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):VMWARE
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):ceph-toolbox

**Comment by travisn** [comment]
@daodaoyuXX Do you see any errors in the mgr pod logs for the crashed module?

**Comment by daodaoyuXX** [comment]
# kubectl -n rook-ceph logs rook-ceph-mgr-a-b966d976b-ztpk9 | grep crash
Defaulted container "mgr" out of: mgr, watch-active, log-collector, chown-container-data-dir (init)
debug 2024-06-27T01:22:11.567+0000 727dfea07200  1 mgr[py] Loading python module 'crash'
debug 2024-06-27T01:22:11.976+0000 727dfea07200 -1 mgr[py] Module crash has missing NOTIFY_TYPES member
debug 2024-06-27T01:22:56.472+0000 76d0056d2200  1 mgr[py] Loading python module 'crash'
debug 2024-06-27T01:22:56.906+0000 76d0056d2200 -1 mgr[py] Module crash has missing NOTIFY_TYPES member

> @daodaoyuXX Do you see any errors in the mgr pod logs for the crashed module?

**Comment by travisn** [comment]
For more details, try `ceph crash ls` and `ceph crash info <crash>`. See #12758 for an example to troubleshoot the mgr module crashes.

**Comment by daodaoyuXX** [comment]
> For more details, try `ceph crash ls` and `ceph crash info <crash>`. See #12758 for an example to troubleshoot the mgr module crashes.

bash-4.4$ ceph crash ls
ID                                                                ENTITY  NEW
2024-07-09T10:04:21.476882Z_9959980e-aa76-43c4-8376-9f5ff98e5d6e  mgr.a    *
2024-07-09T10:04:27.752424Z_e789f0bd-222c-4191-bf3a-1f8be74f6df1  mgr.a    *
2024-07-09T10:04:42.752132Z_ac3a68ef-44d1-4e12-a03d-47534d36729e  mgr.a    *
2024-07-09T10:05:01.166356Z_418fafff-87f6-49a7-8817-9d5590bf7b7d  mgr.a    *
bash-4.4$ crash info 2024-07-09T10:05:01.166356Z_418fafff-87f6-49a7-8817-9d5590bf7b7d
bash: crash: command not found
bash-4.4$ ceph crash info 2024-07-09T10:05:01.166356Z_418fafff-87f6-49a7-8817-9d5590bf7b7d
{
    "backtrace": [
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connection.py\", line 162, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/util/connection.py\", line 80, in create_connection\n    raise err",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/util/connection.py\", line 70, in create_connection\n    sock.connect(sa)",
        "ConnectionRefusedError: [Errno 111] Connection refused",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "Traceback (most recent call last):",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 601, in urlopen\n    chunked=chunked)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 344, in _make_request\n    self._validate_conn(conn)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 844, in _validate_conn\n    conn.connect()",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connection.py\", line 315, in connect\n    conn = self._new_conn()",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connection.py\", line 171, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e)",
        "urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7fdce4bd5550>: Failed to establish a new connection: [Errno 111] Connection refused",
        "\nDuring handling of the above exception, another exception occurred:\n",
        "Traceback (most recent call last):",
        "  File \"/usr/share/ceph/mgr/rook/module.py\", line 108, in available\n    self.k8s.list_namespaced_pod(self._rook_env.namespace)",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/api/core_v1_api.py\", line 12803, in list_namespaced_pod\n    (data) = self.list_namespaced_pod_with_http_info(namespace, **kwargs)  # noqa: E501",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/api/core_v1_api.py\", line 12905, in list_namespaced_pod_with_http_info\n    collection_formats=collection_formats)",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py\", line 345, in call_api\n    _preload_content, _request_timeout)",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py\", line 176, in __call_api\n    _request_timeout=_request_timeout)",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/api_client.py\", line 366, in request\n    headers=headers)",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/rest.py\", line 241, in GET\n    query_params=query_params)",
        "  File \"/usr/lib/python3.6/site-packages/kubernetes/client/rest.py\", line 214, in request\n    headers=headers)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/request.py\", line 68, in request\n    **urlopen_kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/request.py\", line 89, in request_encode_url\n    return self.urlopen(method, url, **extra_kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 324, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 668, in urlopen\n    **response_kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 668, in urlopen\n    **response_kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 668, in urlopen\n    **response_kw)",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 639, in urlopen\n    _stacktrace=sys.exc_info()[2])",
        "  File \"/usr/lib/python3.6/site-packages/urllib3/util/retry.py\", line 399, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))",
        "urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='172.16.0.1', port=443): Max retries exceeded with url: /api/v1/namespaces/rook-ceph/pods (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7fdce4bd5550>: Failed to establish a new connection: [Errno 111] Connection refused',))"
    ],
    "ceph_version": "18.2.2",
    "crash_id": "2024-07-09T10:05:01.166356Z_418fafff-87f6-49a7-8817-9d5590bf7b7d",
    "entity_name": "mgr.a",
    "mgr_module": "rook",
    "mgr_module_caller": "ActivePyModule::dispatch_remote available",
    "mgr_python_exception": "MaxRetryError",
    "os_id": "centos",
    "os_name": "CentOS Stream",
    "os_version": "8",
    "os_version_id": "8",
    "process_name": "ceph-mgr",
    "stack_sig": "aefcd98a439bbccef88302c04e23dac2159872d07c5a99339dfbdf7e98e16777",
    "timestamp": "2024-07-09T10:05:01.166356Z",
    "utsname_hostname": "rook-ceph-mgr-a-6c9c9794bc-hxgx9",
    "utsname_machine": "x86_64",
    "utsname_release": "5.14.0-437.el9.x86_64",
    "utsname_sysname": "Linux",
    "utsname_version": "#1 SMP PREEMPT_DYNAMIC Tue Apr 9 12:57:02 UTC 2024"
}

**Comment by CrazySz** [comment]
this case has solved? if solved,plz tell the root cause and how to sloved,thx

**Comment by CrazySz** [comment]
> bash-4.4$ ceph -v ceph version 18.2.2 (531c0d11a1c5d39fbfe6aa8a521f023abf3bf3e2) reef (stable) bash-4.4$ ceph status cluster: id: c3f35661-a1fb-4858-8afb-1ddb2c462fb1 health: HEALTH_ERR
> 
> [cluster.zip](https://github.com/user-attachments/files/15986240/cluster.zip)
> 
> ```
>         8 mgr modules have recently crashed
> ```
> 
> **File(s) to submit**:
> 
> * Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
> 
> **Logs to submit**:
> 
> * Operator's logs, if necessary
> * Crashing pod(s) logs, if necessary
>   To get logs, use `kubectl -n <namespace> logs <pod name>`
>   When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
>   Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).
> 
> **Cluster Status to submit**:
> 
> * Output of kubectl commands, if necessary
>   To get the health of the cluster, use `kubectl rook-ceph health`
>   To get the status of the cluster, use `kubectl rook-ceph ceph status`
>   For more details, see the [Rook kubectl Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/kubectl-plugin)
>   NAME        DATADIRHOSTPATH   MONCOUNT   AGE     PHASE   MESSAGE                        HEALTH       EXTERNAL   FSID
>   rook-ceph   /var/lib/rook     3          7h13m   Ready   Cluster created successfully   HEALTH_ERR              c3f35661-a1fb-4858-8afb-1ddb2c462fb1
> 
> **Environment**:
> 
> * OS (e.g. from /etc/os-release):Ubuntu 24.04 LTS
> * Kernel (e.g. `uname -a`):GNU/Linux 6.8.0-35-generic x86_64
> * Cloud provider or hardware configuration:
> * Rook version (use `rook version` inside of a Rook Pod):1.14.7
> * Storage backend version (e.g. for ceph do `ceph -v`):ceph version 18.2.2
> * Kubernetes version (use `kubectl version`):v1.30.1
> * Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):VMWARE
> * Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):ceph-toolbox

this case has solved? if solved,plz tell the root cause and how to sloved,thx，i meet the same question

**Comment by CrazySz** [comment]
this case has solved? if solved,plz tell the root cause and how to sloved,thx，i meet the same question

**Comment by am-i-helpful** [comment]
I encountered the same problem in my infra, and analysed the cephcluster (using the command `kubectl describe cephcluster -n rook-ceph`). I tried restarting the rook-ceph operator, which should hopefully refresh the rook-module.

<img width="1275" height="249" alt="Image" src="https://github.com/user-attachments/assets/7bed8c4f-f217-4d74-b70b-155f9fcd0e3b" />

To restart the Rook Ceph operator (which manages the Ceph cluster within Kubernetes), one should just restart the rook-ceph-operator deployment. This can be done using the command `kubectl rollout restart deploy -n rook-ceph rook-ceph-operator`.
